File: leetcode/Arrays_Hash/Single Number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        count = dict()

        for num in nums:
            count[num] = count.get(num, 0) + 1
        
        for num,val in count.items():
            if (val == 1):
                return num
        return 0

File: leetcode/Arrays_Hash/Single Number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map <int, int> count;

        for (int &num : nums){
            count[num]++;
        }
        for (auto num : count){
            if (num.second == 1) return num.first;
        }
        return 0;
    }
};